internal/pkg/skuba/addons/addons.go

```diff
@@ -86,14 +86,16 @@ type Addon struct {
 }
 
 type addonCallbacks interface {
-	beforeApply(AddonConfiguration, *skuba.SkubaConfiguration) error
-	afterApply(AddonConfiguration, *skuba.SkubaConfiguration) error
+	beforeApply(clientset.Interface, AddonConfiguration, *skuba.SkubaConfiguration) error
+	afterApply(clientset.Interface, AddonConfiguration, *skuba.SkubaConfiguration) error
 }
 
 type AddonConfiguration struct {
-	ClusterVersion *version.Version
-	ControlPlane   string
-	ClusterName    string
+	ClusterVersion          *version.Version
+	ControlPlane            string
+	ClusterName             string
+	ControlPlaneHost        string
+	ControlPlaneHostAndPort string
```

Comment:
Why do `ControlPlane`, `ControlPlaneHost` and `ControlPlaneHostAndPort` need to be split out into separate attributes? Couldn't we just infer one from the other?

Reply:
Good suggestion! I'll update it.